src/coreclr/src/jit/lsra.cpp

```diff
@@ -5596,8 +5596,14 @@ void LinearScan::allocateRegisters()
             {
                 allocate = false;
             }
-            else if (refType == RefTypeParamDef && varDsc->lvRefCntWtd() <= BB_UNITY_WEIGHT)
-            {
+            else if (refType == RefTypeParamDef &&
+                     (!currentRefPosition->lastUse || (currentInterval->physReg == REG_STK)) &&
+                     varDsc->lvRefCntWtd() <= BB_UNITY_WEIGHT)
+            {
+                // If this is a low ref-count parameter, and either it is *not* a last used (i.e. unused) or it's
```

Comment:
Nit: I was not clear for me if `not` was applicable to the word in the brackets: `it is *not* a last used (i.e. unused)` should be read as `it is not unused` or `it is unused`? From the context, it looks like the first but maybe then we can delete double negation? 

```suggestion
                // If this is a low ref-count parameter, and either it is used (def is not the last use) or it's
```